### PR TITLE
Add dd-wrt-wifi-devices plugin

### PR DESCRIPTION
**Pull request by cweiske**
.. to monitor the number of connected WiFi devices on a DD-WRT based router

http://cweiske.de/tagebuch/dd-wrt-munin.htm

### CODE REVIEW

File: plugins/dd-wrt/dd-wrt-wifi-devices_ [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+AGPL-3.0-only https://spdx.org/licenses/AGPL-3.0-only.html
+
+=head1 MAGIC MARKERS
+
+#%# family=manual
+
+=cut
+
+if [ -z "$host" ]; then
+    host=`basename $0 | sed 's/dd-wrt-wifi-devices_//g'`
```
Comment (sumpfralle) [inline plugins/dd-wrt/dd-wrt-wifi-devices_:69]: Please use the modern `$(...)` method for command substitution instead of backticks.
In addition please quote `$0` - see the warnings emitted by `shellcheck`.

File: plugins/dd-wrt/dd-wrt-wifi-devices_ [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+fi
+user=${user:-root}
+pass=${password:-admin}
+
+case $1 in
+   config)
+       echo "graph_title DD-WRT $host connected Wifi devices\n";
```
Comment (sumpfralle) [inline plugins/dd-wrt/dd-wrt-wifi-devices_:76]: Th `\n` is not necessary, or?
(at least my bash's `echo` emits it literally)
And please remove the optional semicolon.

File: plugins/dd-wrt/dd-wrt-wifi-devices_ [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+       echo "graph_vlabel Number of devices\n";
+       echo "graph_args --base 1000\n";
+       echo "graph_category network\n";
+       echo "graph_order wifi\n";
+       echo "wifi.label Wifi Connections\n";
+       echo "wifi.type GAUGE\n";
+       echo "wifi.graph LINE1\n";
+       #echo "wifi.info Wifi Connections\n";
```
Comment (sumpfralle) [inline plugins/dd-wrt/dd-wrt-wifi-devices_:84]: Please remove such commented code.

File: plugins/dd-wrt/dd-wrt-wifi-devices_ [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+url="http://$host/Status_Wireless.live.asp"
+
+count=$(
+    curl --silent --show-error --fail --user "$user:$pass" --max-time 3 "$url"\
+        | grep -o 'assoc_count::[0-9]*'\
+        | cut -d':' -f 3
+)
+#without "-o pipefail" we would not get curl's exit code here
+exitcode=$?
+if [ "$exitcode" -gt 0 ]; then
```
Comment (sumpfralle) [inline plugins/dd-wrt/dd-wrt-wifi-devices_:100]: In this case you should emit `U` ("unkown") as the value.

File: plugins/dd-wrt/dd-wrt-wifi-devices_ [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    use_node_name no
+
+
+=head1 REQUIREMENTS
+
+"bash" and "curl" are required.
+
+=head1 SEE ALSO
+
+https://github.com/Tafkas/fritzbox-munin
```
Comment (sumpfralle) [inline plugins/dd-wrt/dd-wrt-wifi-devices_:53]: Is this repository really related to this specific plugin?

**Review by sumpfralle — COMMENTED** [review]
Thank you for contributing this plugin!

Please take a look at my comments below.

Just for your information: there is also the `muninlite` package for OpenWrt, which can be used for collecting data from within an OpenWrt router. I am not sure, whether this is applicable for DD-Wrt, as well. But this is not related to this plugin proposal.

File: plugins/dd-wrt/dd-wrt-wifi-devices_ (hunk 2 above)
Comment (cweiske) [reply plugins/dd-wrt/dd-wrt-wifi-devices_]: This was a leftover of the original PHP version. Fixed.

File: plugins/dd-wrt/dd-wrt-wifi-devices_ (hunk 3 above)
Comment (cweiske) [reply plugins/dd-wrt/dd-wrt-wifi-devices_]: Removed.

File: plugins/dd-wrt/dd-wrt-wifi-devices_ (hunk 4 above)
Comment (cweiske) [reply plugins/dd-wrt/dd-wrt-wifi-devices_:99]: Fixed.

**Comment by cweiske** [comment]
I've fixed the issues you mentioned @sumpfralle. Thanks for the review.

**Comment by sumpfralle** [comment]
Thank you!
